src/test/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryListFilterChainTest.java

```diff
@@ -322,4 +329,35 @@ public void removeTransitWithHigherCostThanBestOnStreetOnlyEnabled() {
       assertEquals(toStr(List.of(walk)), toStr(chain.filter(List.of(walk, bus))));
     }
   }
+
+  @Nested
+  class AddEmissionsToItineraryTest {
+
+    Itinerary bus;
+    Itinerary car;
+    ItineraryListFilterChainBuilder builder = createBuilder(true, false, 2);
+
+    @BeforeEach
+    public void setUpItineraries() {
+      bus = newItinerary(A).bus(21, T11_06, T11_09, B).build();
+      car = newItinerary(A).drive(T11_30, T11_50, B).build();
+    }
+
+    @Test
+    public void emissionsTest() {
+      var emissionsService = Mockito.mock(EmissionsService.class);
```

Comment:
I wonder if we could write tests for this without using Mockito.

Reply:
Sure, I changed it to use actual classes and test that values are present.